tools/ci_build/github/linux/build_linux_arm64_python_package.sh

```diff
@@ -1,26 +1,51 @@
-#/bin/bash
-set -x
+#!/bin/bash
+set -e -x
 mkdir /build/dist
-CFLAGS="-Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -fstack-clash-protection -O3 -Wl,--strip-all" 
-CXXFLAGS="-Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -fstack-clash-protection -O3 -Wl,--strip-all"
-arch=$(uname -i)
-if [[ $arch == x86_64* ]]; then
+CFLAGS="-Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -O3 -pipe -Wl,--strip-all"
+CXXFLAGS="-Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -O3 -pipe -Wl,--strip-all"
+
+
+BUILD_DEVICE="CPU"
+BUILD_CONFIG="Release"
+while getopts "d:" parameter_Option
+do case "${parameter_Option}"
+in
+#GPU or CPU. 
+d) BUILD_DEVICE=${OPTARG};;
+esac
+done
+
+GCC_VERSION=$(gcc -dumpversion)
+#-fstack-clash-protection prevents attacks based on an overlapping heap and stack.
+if [ "$GCC_VERSION" -ge 8 ]; then
+    CFLAGS="$CFLAGS -fstack-clash-protection"
+    CXXFLAGS="$CXXFLAGS -fstack-clash-protection"
+fi
+
+ARCH=$(uname -i)
+
+if [ $ARCH == "x86_64" ] && [ "$GCC_VERSION" -ge 9 ]; then
     CFLAGS="$CFLAGS -fcf-protection"
-    CXXFLAGS="$CFLAGS -fcf-protection"
+    CXXFLAGS="$CXXFLAGS -fcf-protection"
+fi
+
+
+BUILD_ARGS="--build_dir /build --config $BUILD_CONFIG --update --build --skip_submodule_sync --parallel --enable_lto --build_wheel"
+
+if [ $ARCH == "x86_64" ]; then
+    #ARM build machines do not have the test data yet.
+    BUILD_ARGS="$BUILD_ARGS --enable_onnx_tests"
+fi
+if [ $BUILD_DEVICE == "GPU" ]; then
+    BUILD_ARGS="$BUILD_ARGS --use_cuda --use_tensorrt --cuda_version=11.6 --tensorrt_home=/usr --cuda_home=/usr/local/cuda-11.6 --cudnn_home=/usr/local/cuda-11.6"
```

Comment:
This PR forgot to set 
" --cmake_extra_defines 'CMAKE_CUDA_ARCHITECTURES=37;50;52;60;61;70;75;80'"

I will submit another PR to fix it.